contrib/ruby/ext/trilogy-ruby/cast.c

```diff
@@ -144,8 +144,12 @@ rb_trilogy_cast_value(const trilogy_value_t *value, const struct column_info *co
         case TRILOGY_TYPE_NEWDECIMAL: {
             // TODO - optimize so we don't have to allocate a ruby string for
             // decimal columns
-            VALUE str = rb_str_new(value->data, value->data_len);
-            return rb_funcall(rb_mKernel, id_BigDecimal, 1, str);
+            if (column->decimals == 0) {
+              return rb_cstr2inum(value->data, 10);
+            } else {
+              VALUE str = rb_str_new(value->data, value->data_len);
```

Comment:
This file uses 4 space indent.

Also tiny, but as you're touching it anyway, the above TODO can come down into this else clause -- in the integer case, with this change it's to-done.